cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -418,8 +418,8 @@ void reader_impl::decode_page_data(read_mode mode,
   // that it is difficult/impossible for a given page to know that it is writing the very
   // last value that should then be followed by a terminator (because rows can span
   // page boundaries).
-  std::vector<size_type*> out_buffers;
-  std::vector<size_type> final_offsets;
+  auto out_buffers   = cudf::detail::make_host_vector<size_type*>(0, _stream);
+  auto final_offsets = cudf::detail::make_host_vector<size_type>(0, _stream);
```

Comment:
Note: `cudf::detail::host_vector` should work well as [`thrust::host_vector`](https://nvidia.github.io/cccl/thrust/api/classthrust_1_1host__vector.html) when using the cudf/RMM memory allocator.

The reason I raise this is that most existing uses of `host_vector` in cudf treat it as a fixed-size array. In contrast, this particular case starts with zero-sized and relies on dynamic resizing